src/operators.jl

```diff
@@ -851,3 +851,76 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+"""
+    mycielski(g)
+
+Returns a graph after applying the Mycielski operator to the input. The Mycielski operator
+returns a new graph with `2n+1` vertices and `3e+n` edges and will increase the chromatic
+number of the graph by 1.
+
+The Mycielski operation can be repeated by using the `iterations` kwarg. Each time, it will
+apply the operator to the previous iterations graph.
+
+For each vertex in the original graph, that vertex and a copy of it is added to the new graph.
+Then, for each edge `(x, y)` in the original graph, the edges `(x, y)`, `(x', y)`, and `(x, y')`
+are added to the new graph, where `x'` and `y'` are the "copies" of `x` and `y`, respectively.
+In otherwords, the original graph is present as a subgraph, and each vertex in the original graph
+is connected to all of it's neighbors' copies. Finally, one last vertex `w` is added to the graph
+and an edge connecting each copy vertex `x'` to `w` is added.
+
+See [Mycielskian](https://en.wikipedia.org/wiki/Mycielskian) for more information.
+
+# Examples
+```jldoctest
+julia> c = CycleGraph(5)
+{5, 5} undirected simple Int64 graph
+
+julia> m = Graphs.mycielski(c)
+{11, 20} undirected simple Int64 graph
+
+julia> collect(edges(m))
+20-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
+ Edge 1 => 2
+ Edge 1 => 5
+ Edge 1 => 7
+ Edge 1 => 10
+ Edge 2 => 3
+ Edge 2 => 6
+ Edge 2 => 8
+ Edge 3 => 4
+ Edge 3 => 7
+ Edge 3 => 9
+ Edge 4 => 5
+ Edge 4 => 8
+ Edge 4 => 10
+ Edge 5 => 6
+ Edge 5 => 9
+ Edge 6 => 11
+ Edge 7 => 11
+ Edge 8 => 11
+ Edge 9 => 11
+ Edge 10 => 11
+```
+"""
+@traitfn function mycielski(g::AbstractGraph::(!IsDirected); iterations = 1)
+    ref = g
+    out = deepcopy(g)
```

Comment:
I think that `deepcopy(g)` could slow down and uses a lot of memory especially for big graphs.

Reply:
These operators are not supposed to mutate the input right? I see some have `!` variants, but not all. Perhaps there should be a `mycielski!` and `mycielski(g; iterations) = mycielski!(deepcopy(g); iterations=iterations)` method. WDYT?

So this operator will make at least a 2x larger graph so at least a deep copy's worth of data has to be initialized, and the deep copy allows us to build it in place (in the event of `iterations > 1`, and some other operators also use deep copy. Maybe a shallow copy is ok, but I am not quite sure of the graph internals.

Reply:
Yes I agree with you. I was only thinking about the case in which we can mutate the input.

Reply:
I would also support having both `mycielski!` and `mycielski`. We currently have something similar for `transitiveclosure` and `transitiveclosure!` at the moment.